src/provider/provider_tracking.c

```diff
@@ -76,6 +136,64 @@ static umf_result_t umfMemoryTrackerAdd(umf_memory_tracker_handle_t hTracker,
     return UMF_RESULT_ERROR_UNKNOWN;
 }
 
+// Get the most nested (on the highest level) entry in the map with the `ptr` key.
+// If `no_children` is set to 1, the function will return the entry
+// only if it has no children on the higher level.
+// The function returns the entry if found, otherwise NULL.
+static tracker_alloc_info_t *
+map_get_most_nested_entry(umf_memory_tracker_handle_t hTracker, const void *ptr,
+                          int *_level, uintptr_t *_parent_key,
+                          tracker_alloc_info_t **_parent_value,
+                          int no_children) {
+    assert(ptr);
+
+    tracker_alloc_info_t *parent_value = NULL;
+    tracker_alloc_info_t *rvalue = NULL;
+    uintptr_t parent_key = 0;
+    uintptr_t rkey = 0;
+    int level = 0;
+    int found = 0;
+
+    do {
+        found =
```

Comment:
I understand that according to current logic the `level` cannot go higher then `MAX_LEVELS_OF_MAPS` but let's put a debug assert just to check that we did not make a mistake.

Reply:
Done.